File: SharedLibrary/Shared/UserRoles.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SharedLibrary.Shared
{
	public static class UserRoles
	{
		public const string Teacher = "Teacher";
		public const string Administrator = "Administrator";
		public const string Student = "Student";
		public const string Tester = "Tester";
		public const string Public = "Public";
		public const string FakeStudent = "StudentFake";
		public const string Support = "Support";
		public const string HiddenStudent = "StudentHidden";
		public const string Robot = "Robot";
		public const string Developer = "Developer";
		public const string LoginOnly = "LoginOnly";
		public const string SchoolManager = "SchoolManager";

		// public const string SupportOrAdmin = Support + ", " + Administrator;
		public const string AdminOrDeveloper = Administrator + ", " + Developer;
		public const string AdminOrDeveloperOrTeacher = AdminOrDeveloper + ", " + SchoolManager + ", " + Teacher;
	}
}

File: MobileApp/zoliky_teachers/lib/utils/api/enums/UserRoles.dart
import 'package:zoliky_teachers/utils/api/enums/Enum.dart';

class UserRole extends Enum<String> {
  const UserRole(String value) : super(value);

  static const UserRole Teacher = const UserRole("Teacher");
  static const UserRole Administrator = const UserRole("Administrator");
  static const UserRole Student = const UserRole("Student");
  static const UserRole Tester = const UserRole("Tester");
  static const UserRole Public = const UserRole("Public");
  static const UserRole FakeStudent = const UserRole("StudentFake");
  static const UserRole Support = const UserRole("Support");
  static const UserRole HiddenStudent = const UserRole("StudentHidden");
  static const UserRole Robot = const UserRole("Robot");
  static const UserRole Developer = const UserRole("Developer");
  static const UserRole LoginOnly = const UserRole("LoginOnly");
  static const UserRole SchoolManager = const UserRole("SchoolManager");

  operator ==(Object other) => (other is UserRole && other.value == this.value);

  int get hashCode => value.hashCode;
}
